drivers/osdp/src/osdp_common.h

```diff
@@ -0,0 +1,404 @@
+	 * @brief This capability indicates whether the PD supports the
+	 * transparent mode used for communicating directly with a smart card.
+	 */
+	CAP_SMART_CARD_SUPPORT,
+
+	/**
+	 * @brief This capability indicates the number of credential reader
+	 * devices present. Compliance levels are bit fields to be assigned as
+	 * needed.
+	 */
+	CAP_READERS,
+
+	/**
+	 * @brief This capability indicates the ability of the reader to handle
+	 * biometric input
+	 */
+	CAP_BIOMETRICS,
+
+	/**
+	 * @brief Capability Sentinel
+	 */
+	CAP_SENTINEL
+};
+
+struct osdp_cmd_queue {
+	struct osdp_cmd *front;
+	struct osdp_cmd *back;
+};
+
+/**
+ * @brief PD capability structure. Each PD capability has a 3 byte
+ * representation.
+ *
+ * @param function_code One of enum osdp_pd_cap_function_code_e.
+ * @param compliance_level A function_code dependent number that indicates what
+ *                         the PD can do with this capability.
+ * @param num_items Number of such capability entities in PD.
+ */
+struct pd_cap {
+	uint8_t function_code;
+	uint8_t compliance_level;
+	uint8_t num_items;
+};
+
+/**
+ * @brief PD ID information advertised by the PD.
+ *
+ * @param version 3-bytes IEEE assigned OUI
+ * @param model 1-byte Manufacturer's model number
+ * @param vendor_code 1-Byte Manufacturer's version number
+ * @param serial_number 4-byte serial number for the PD
+ * @param firmware_version 3-byte version (major, minor, build)
+ */
+struct pd_id {
+	int version;
+	int model;
+	uint32_t vendor_code;
+	uint32_t serial_number;
+	uint32_t firmware_version;
+};
+
+struct osdp_channel {
+	/**
+	 * @brief pointer to a block of memory that will be passed to the
+	 * send/receive method. This is optional and can be left empty.
+	 */
+	void *data;
+
+	/**
+	 * @brief pointer to function that copies received bytes into buffer
+	 * @param data for use by underlying layers. channel_s::data is passed
+	 * @param buf byte array copy incoming data
+	 * @param len sizeof `buf`. Can copy utmost `len` bytes into `buf`
+	 *
+	 * @retval +ve: number of bytes copied on to `bug`. Must be <= `len`
+	 * @retval -ve on errors
+	 */
+	int (*recv)(void *data, uint8_t *buf, int maxlen);
+
+	/**
+	 * @brief pointer to function that sends byte array into some channel
+	 * @param data for use by underlying layers. channel_s::data is passed
+	 * @param buf byte array to be sent
+	 * @param len number of bytes in `buf`
+	 *
+	 * @retval +ve: number of bytes sent. must be <= `len`
+	 * @retval -ve on errors
+	 */
+	int (*send)(void *data, uint8_t *buf, int len);
+
+	/**
+	 * @brief pointer to function that drops all bytes in TX/RX fifo
+	 * @param data for use by underlying layers. channel_s::data is passed
+	 */
+	void (*flush)(void *data);
+};
+
+struct osdp_pd {
+	void *__parent;
+	int offset;
+	uint32_t flags;
+
+	/* OSDP specified data */
+	int baud_rate;
+	int address;
+	int seq_number;
+	struct pd_cap cap[CAP_SENTINEL];
+	struct pd_id id;
+
+	/* PD state management */
+	int state;
```

Comment:
About various state variables here, could you use enum's instead?

Reply:
Done.